libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt

```diff
@@ -926,8 +926,12 @@ class MapboxNavigation @VisibleForTesting internal constructor(
         // Telemetry uses this field to determine what type of event should be triggered.
         val setRoutesInfo = when {
             routes.isEmpty() -> SetRoutes.CleanUp
-            routes.first() == directionsSession.routes.firstOrNull() ->
+            routes.first() == directionsSession.routes.firstOrNull() -> {
                 SetRoutes.Alternatives(initialLegIndex)
+            }
+            routes.sortedBy { it.id } == directionsSession.routes.sortedBy { it.id } -> {
```

Comment:
You said that [A, B, C] -> [B, A] should result in alternatives (internal reorder) reason. But looks like now it'll be new.

Reply:
Fixed according to new discsussion with NN team.